gnucash/CMakeLists.txt

```diff
@@ -42,6 +43,46 @@ TARGET_LINK_LIBRARIES (gnucash
    ${GUILE_LDFLAGS} ${GLIB2_LDFLAGS} ${GTK3_LDFLAGS} ${GTK_MAC_LDFLAGS}
 )
 
+# Get glib executable for generating the gresource file
+EXECUTE_PROCESS(
+   COMMAND
+       ${PKG_CONFIG_EXECUTABLE} gio-2.0 --variable glib_compile_resources
+   OUTPUT_VARIABLE
+       GLIB_COMPILE_RESOURCES_EXECUTABLE
+   OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Get the dependencies of the gresource
+EXECUTE_PROCESS(
+  OUTPUT_VARIABLE
+      gr_files
+  WORKING_DIRECTORY
+      ${CMAKE_CURRENT_SOURCE_DIR}
+  COMMAND ${GLIB_COMPILE_RESOURCES_EXECUTABLE}
+     --generate-dependencies
+     ${CMAKE_CURRENT_SOURCE_DIR}/gnucash-gresources.xml
+)
+
+STRING (REPLACE "\n" ";" gresource_files ${gr_files})
+
+SET (DEPENDS "")
+FOREACH (cur_file ${gresource_files})
+    LIST (APPEND gnucash_GRESOURCE "${cur_file}")
+ENDFOREACH ()
+
+ADD_CUSTOM_COMMAND(
+   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/gnucash-gresources.c
+   COMMAND
+    ${GLIB_COMPILE_RESOURCES_EXECUTABLE}
+    --target=${CMAKE_CURRENT_BINARY_DIR}/gnucash-gresources.c
+    --sourcedir=${CMAKE_CURRENT_SOURCE_DIR}
+    --generate-source
+    gnucash-gresources.xml
```

Comment:
It would make slightly more sense to me to run this command with WORKING DIRECTORY set to ${CMAKE_CURRENT_BINARY_DIR} (file generation should generally happen there as much as possible).
You can then probably omit most ${CMAKE_CURRENT_BINARY_DIR}/ prefixes (including in the OUTPUT line) and need to add a ${CMAKE_CURRENT_SOURCE_DIR}/ prefix to gnucash-gresources.xml.